Assets/MixedRealityToolkit.SDK/Profiles/DefaultMixedRealityGesturesProfile.asset

```diff
@@ -36,3 +37,9 @@ MonoBehaviour:
       id: 9
       description: Manipulate Action
       axisConstraint: 0
+  - description: Tap
```

Comment:
A few of the historical reasons we didn't add tap to the gesture recognizer in v2:
1. Not all platforms support a specifically defined "tap".
1. We could define "tap" the way we define "click" today for those platforms, while specially defining "click/tap" for WMR.
    a. This is interesting, because then we have a better defined click on WMR while still supporting click on other platforms
    b. but then we run into the potential of non-deterministic ordering of events, especially across platforms, since we'll be listening to one polling-based implementation for "down", "up", and (on some platforms) "click", and an event-based implementation for "click/tap", on only one platform.

Reply:
Yeah, this makes sense. I think your point in 2.a. about having a "better defined click" is particularly compelling to me - as my work is focused on WMR, "native" behavior is preferable to "consistent cross-platform" behavior. I appreciate that the latter is important for MRTK and I think that has to be the first priority, but I'd like to get both if we can.
Regarding 2.b. - that seems totally plausible and I can imagine it causing problems. I assume if we are just making up the events ourselves (as in UnityTouchController) we can make it match, but I wouldn't be that surprised if other platforms have defined "clicks" that don't fully agree with WMR.

Reply:
@keveleigh I guess in the end all I really care about is that there is some event (click/tap/select) on input up that is mutually exclusive with hold/manipulate. I don't think there's any particular reason it needs to come from the gesture recognizer - e.g. I'd also be ok with having the pointer decide what constitutes a click vs hold vs drag (as in UnityTouchController).

At the same time, I also kind of feel like maybe it's ok to use the platform implementations as long as it is understood that the exact ordering is not well-defined. I don't recall offhand which of tap or input up comes first (or even if it is necessarily consistent), and as it logically seems like they should happen at the same time I would try to avoid writing code that assumes a certain ordering. I also appreciate that such assumptions often happen unintentionally - but that seems unavoidable to some extent when dealing with heterogeneous controllers. E.g. does MRTK make any guarantees about the timing of the hold/manipulate gestures? Or just assume that they are "close enough"?